src/server/routes/consents.ts

```diff
@@ -365,6 +367,10 @@ router.post(
         state.queryParams,
       );
 
+      setEncryptedStateCookie(res, {
```

Comment:
I'd suggest using the `updateEncryptedStateCookie` method instead, as this merges (updates/upserts) the state rather than overwriting it which is what `set` does.